### PR TITLE
PR: Fix several indentation cases

### LINKED ISSUE #11128: Indentation level is not reset when writing a comment after indent within a long instruction
### Problem Description

When breaking up a long instruction into several lines to comply with the style guide then indentation is necessary.

After entering the instruction and hitting enter the cursor will be at the last used indentation level. Hitting backspace resets that for this and future empty lines and lines of code, however not for following comment lines.

I have switched off the option "Automatic indentation after 'else', 'elif', etc." and it happens nonetheless.

### What steps reproduce the problem?

After entering the test variable in a file in the editor hit enter and backspace to reset the indentation level to zero, then enter the following lines manually:

This works fine:
```
this_is_a_test_variable\
    = ["aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"]
a = 1
# this comment is fine, hit enter for next line
# this comment is fine too
```

This does not:
```
this_is_a_test_variable\
    = ["aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"]
# this comment was written after resetting indentation, hit enter for next line
    # this line is indented again
```

## Versions
* Spyder version: 4.0.0
* Python version: 3.7.3
* Qt version: 5.12.5
* PyQt version: 5.12.3
* Operating System name/version: Windows 10

### Dependencies
```
cloudpickle >=0.5.0          :  1.2.2 (OK)
pygments >=2.0               :  2.5.2 (OK)
qtconsole >=4.6.0            :  4.6.0 (OK)
nbconvert >=4.0              :  5.6.1 (OK)
sphinx >=0.6.6               :  2.2.2 (OK)
pylint >=0.25                :  2.4.4 (OK)
psutil >=0.3                 :  5.6.7 (OK)
qtawesome >=0.5.7            :  0.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
zmq >=17                     :  18.1.1 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
numpydoc >=0.6.0             :  0.9.1 (OK)
spyder_kernels >=1.8.1;<2.0.0:  1.8.1 (OK)
qdarkstyle >=2.7             :  2.7 (OK)
atomicwrites >=1.2.0         :  1.3.0 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
intervaltree                 :  None (OK)
watchdog                     :  None (OK)
keyring                      :  None (OK)
pexpect >=4.4.0              :  4.7.0 (OK)
pympler                      :  None (OK)
sympy >=0.7.3                :  1.4 (OK)
cython >=0.21                :  0.29.14 (OK)
IPython >=4.0                :  7.10.1 (OK)
matplotlib >=2.0.0           :  3.1.2 (OK)
pandas >=0.13.1              :  0.25.3 (OK)
numpy >=1.7                  :  1.17.3 (OK)
scipy >=0.17.0               :  1.3.1 (OK)
pyls >=0.31.2;<0.32.0        :  0.31.2 (OK)
rtree >=0.8.3                :  0.9.1 (OK)

```

### PR DESCRIPTION
**Pull request by impact27**
## Description of Changes
Fix indentation:
 - after a comment
 - when doing backspace with a blank line
 - search a maximum of two lines up when looking for indentation

* [x] Wrote at least one-line docstrings (for any new functions)
* [x] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #11128

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

**Comment by ccordoba12** [comment]
@dalthviz, please review this one.

File: spyder/plugins/editor/widgets/codeeditor.py [hunk 1]
```diff
@@ -2959,8 +2974,15 @@ def fix_indent_smart(self, forward=True, comment_or_string=False):
 
         # TODO untangle this block
         if prevline and not bracket_stack and not prevtext.endswith(':'):
-            cur_indent = self.get_block_indentation(block_nb - 1)
-            is_blank = not self.get_text_line(block_nb - 1).strip()
+            if forward:
+                # Keep indentation of previous line
+                ref_line = block_nb-1
```
Comment (dalthviz) [inline spyder/plugins/editor/widgets/codeeditor.py]: Please, add spaces around the operator here

File: spyder/plugins/editor/widgets/codeeditor.py [hunk 2]
```diff
@@ -3882,14 +3908,14 @@ def _handle_completions(self):
                     self._last_key_pressed_text = ''
                     self._last_pressed_key = None
 
-    def fix_and_strip_indent(self, comment_or_string=False):
+    def fix_and_strip_indent(self, *args, **kwargs):
```
Comment (dalthviz) [inline spyder/plugins/editor/widgets/codeeditor.py]: Maybe we could specify the `kwargs` description in the docstring to be clear about the possible `kwargs` the method receives?

**Comment by ccordoba12** [comment]
@dalthviz, please approve when you think this is ready.

**Pull request merged**